### ISSUE TITLE
Please expose `#query()` function on `MetadataReader` class - or at least, expose a function which retrieves the Entity ID

### ISSUE THREAD

**Issue by oliverlockwood** [compwright/passport-saml-metadata#40]
We have been happily using this library for a while. Thanks for sharing it with the community, and for keeping it maintained.

We support passport-saml's `MultiSamlStrategy`, using the EntityID in the SAML Metadata as a distinguishing tool between different IdPs.

In order to get hold of the Entity ID, we have this little block of code:

```typescript
    private static extractEntityId(metadataReader: MetadataReader): string {

        let entityId: string;

        try {
            entityId = metadataReader.query('//md:EntityDescriptor/@entityID')[0].nodeValue;
        }
        catch (err) {
            // N.B. the library debug logs in error cases
            entityId = metadataReader.query('//EntityDescriptor/@entityID/')[0].nodeValue;
        }

        return entityId;
    }
```

The catch clause may or may not be overly defensive, but that's not the point here 😄 

I tried upgrading to use the newly released version `3.0.0` of this library - as part of upgrading to use the new co-ordinates of `passport-saml` - but unfortunately I was unable to succeed, because the `#` prefix that was added to the `query()` function makes it inaccessible.

Would you be willing to consider exposing this query function again?  Or if not, at least to add a new wrapper function which uses it to retrieves the `'//md:EntityDescriptor/@entityID'` ?

I would be happy to contribute a PR, but I thought I'd first seek your approval in principle and guidance over which path you'd prefer.

**Comment by compwright** [comment]
@oliverlockwood thank you for the feedback, I'm glad you're finding this library useful!

I'd love a PR.

I am partial to adding a getter method to retrieve the entity ID versus exposing query, which was never part of the documented API.

**Comment by oliverlockwood** [comment]
Good stuff, thanks @compwright for the positivity and the guidance.  PR #41 now raised.

Whenever we get to the point of my changes being accepted, I'd be grateful if you would issue a new release of the library so that I can make use of this new functionality.  Thanks again, and Merry Christmas!

**Comment by compwright** [comment]
Fixed in v3.1.0 - thank you for this contribution!

**Comment by oliverlockwood** [comment]
My pleasure! Thank you.
